File: code/functions/function-type.kt
// Function types in Kotlin
fun add(a: Int, b: Int): Int = a + b
fun multiply(a: Int, b: Int): Int = a * b

fun applyOperation(op: (Int, Int) -> Int, x: Int, y: Int): Int = op(x, y)

fun main() {
    var operation: (Int, Int) -> Int = ::add
    var result = applyOperation(operation, 5, 3)
    println("Result: $result")
    
    operation = ::multiply
    result = applyOperation(operation, 5, 3)
    println("Result: $result")
}

File: code/functions/function-type.ex
# Function types in Elixir
add = fn(a, b) -> a + b end
multiply = fn(a, b) -> a * b end

apply_operation = fn(op, x, y) -> op.(x, y) end

operation = add
result = apply_operation.(operation, 5, 3)
IO.puts "Result: #{result}"

operation = multiply
result = apply_operation.(operation, 5, 3)
IO.puts "Result: #{result}"